posts/MarkoB/2025-12-15-hibernate-search-8-2-0-Final.adoc

```diff
@@ -0,0 +1,140 @@
+* link:{hsearch-jira-url-prefix}/HSEARCH-5536[HSEARCH-5536]: Upgrade to Elasticsearch **client** 9.2.2 (`elasticsearch-rest5-client`)
+* link:{hsearch-jira-url-prefix}/HSEARCH-5524[HSEARCH-5524]: Upgrade to OpenSearch **client** 3.3.2 (`opensearch-rest-client`)
+* link:{hsearch-jira-url-prefix}/HSEARCH-5540[HSEARCH-5540]: Upgrade to AWS SDK 2.40.4
+* link:{hsearch-jira-url-prefix}/HSEARCH-5506[HSEARCH-5506]: Upgrade to Avro 1.12.1
+
+[[pluggable-clients]]
+=== Pluggable Elasticsearch backend REST clients
+
+This version of Hibernate Search introduces pluggable REST clients for the Elasticsearch backend.
+Historically, the Elasticsearch backend relied on the Elasticsearch low-level REST client,
+which is based on the Apache HTTP Client 4. With the new pluggable client architecture,
+it is possible to switch to a different REST client as needed.
+This may be useful when you are running Hibernate Search in an environment relying on a different HTTP client type (e.g. Vert.x).
+or when you need to use some distribution specific client features and for example need to switch to the OpenSearch REST client.
+
+Hibernate Search provides the following REST client options out of the box:
+
+[[backend-elasticsearch-configuration-client-elasticsearch-client-rest4]]
+==== Elasticsearch low level REST client (Apache HTTP Client 4)
+
+Coordinates::
+GroupID::: `org.hibernate.search`
+ArtifactID::: `hibernate-search-backend-elasticsearch-client-rest4`
+Underlying HTTP Client::: Apache HTTP Client 4
+
+This Elasticsearch backend REST client is based on the Elasticsearch low level client (`org.elasticsearch.client:elasticsearch-rest-client`).
+Currently, the default REST client used by the Elasticsearch backend.
+
+[[backend-elasticsearch-configuration-client-elasticsearch-client-rest5]]
+==== Elasticsearch low level REST client (Apache HTTP Client 5)
+
+Coordinates::
+GroupID::: `org.hibernate.search`
+ArtifactID::: `hibernate-search-backend-elasticsearch-client-rest5`
+Underlying HTTP Client::: Apache HTTP Client 5
+
+This Elasticsearch backend REST client is based on the newer Elasticsearch low level client (`co.elastic.clients:elasticsearch-rest5-client`),
+which is also used in the `co.elastic.clients:elasticsearch-java`.
+
+[[backend-elasticsearch-configuration-client-elasticsearch-client-opensearch]]
+==== OpenSearch low level REST client (Apache HTTP Client 5)
+
+Coordinates::
+GroupID::: `org.hibernate.search`
+ArtifactID::: `hibernate-search-backend-elasticsearch-client-opensearch-rest`
+Underlying HTTP Client::: Apache HTTP Client 5
+
+This Elasticsearch backend REST client is based on the OpenSearch low level client (`org.opensearch.client:opensearch-rest-client`).
+
+NOTE: It is recommended to explicitly add a client that you prefer as a dependency, since in the future version
+of Hibernate Search we will remove the implicit dependency on `hibernate-search-backend-elasticsearch-client-rest4`
+from the Elasticsearch backend.
```

Comment:
This shouldn't be in this section, I think. It's not really relevant for `opensearch-rest`, because people using that will obviously need to add a dependency.

Either put this in the `rest4` section (because it's mostly useful to `rest4` lovers), or under "Pluggable Elasticsearch backend REST clients".